pkg/util/kubernetes/apiserver/services.go

```diff
@@ -14,13 +14,40 @@ import (
 	"k8s.io/api/core/v1"
 )
 
-// mapServices maps each pod (endpoint) to the metadata associated with it.
-// It is on a per node basis to avoid mixing up the services pods are actually connected to if all pods of different nodes share a similar subnet, therefore sharing a similar IP.
-func (metaBundle *MetadataMapperBundle) mapServices(nodeName string, pods v1.PodList, endpointList v1.EndpointsList) error {
-	metaBundle.m.Lock()
-	defer metaBundle.m.Unlock()
+// ServicesMapper maps pod names to the names of the services targeting the pod
+// keyed by the namespace a pod belongs to. This data structure allows for O(1)
+// lookups of services given a namespace and pod name.
+//
+// The data is stored in the following schema:
+// {
+// 	"namespace": {
+// 		"pod": [ "svc1", "svc2", "svc3" ]
+// 	}
+// }
+type ServicesMapper map[string]map[string][]string
```

Comment:
This is still a complex type but at least gives a more descriptive interface to it with some error checking.

Reply:
Agree